go-controller/pkg/ovn/egressip.go

```diff
@@ -1823,52 +1889,42 @@ func (e *egressIPZoneController) deleteStaticRouteOps(ops []ovsdb.Operation, pod
 // It also returns the list of podIPs whose routes and SNAT's were deleted
 // This function should be called with a lock on e.nodeZoneState.status.Node
 func (e *egressIPZoneController) deleteEgressIPStatusSetup(name string, status egressipv1.EgressIPStatusItem) ([]net.IP, error) {
-	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
-	var nextHopIP string
 	var err error
 	isLocalZoneEgressNode, loadedEgressNode := e.nodeZoneState.Load(status.Node)
-	if loadedEgressNode && isLocalZoneEgressNode {
-		gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-		if errors.Is(err, libovsdbclient.ErrNotFound) {
-			// if the gateway router join IP setup is already gone, then don't count it as error.
-			klog.Warningf("Unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
-		} else if err != nil {
-			return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
-		}
-		if gatewayRouterIP != nil {
-			nextHopIP = gatewayRouterIP.String()
-		}
-	} else if config.OVNKubernetesFeature.EnableInterconnect {
-		// fetch node annotation of the egress node
-		nextHopIP, err = e.getTransitIP(status.Node, isEgressIPv6)
-		if apierrors.IsNotFound(err) {
-			// if the node is already gone, then don't count it as error as during deletion.
-			// technically speaking, the egressNode handler should always get the delete event first before node handler
-			klog.Warningf("Unable to fetch transit switch IP for node: %s: err: %v", status.Node, err)
-		} else if err != nil {
-			return nil, fmt.Errorf("unable to fetch transit switch IP for node %s: %w", status.Node, err)
+
+	var ops []ovsdb.Operation
+	var nextHopIP string
+	eNode, err := e.watchFactory.GetNode(status.Node)
+	if err == nil {
+		eIPIP := net.ParseIP(status.EgressIP)
+		isOVNManagedNetwork, err := util.IsOVNManagedNetwork(eNode, eIPIP)
+		if err == nil {
```

Comment:
We should at least log the errors if they are not nil

Reply:
done